src/main/java/io/dropwizard/sharding/dao/LookupDao.java

```diff
@@ -430,6 +428,39 @@ public LockedContext(int shardId, SessionFactory sessionFactory, Function<T, T>
             });
         }
 
+        public<U> LockedContext<T> selectAndUpdateOrSave(RelationalDao<U> relationalDao, DetachedCriteria criteria, Function<U, U> handler) {
+            return apply(parent-> {
+                try {
+                    relationalDao.selectAndUpdateOrSave(this, criteria, handler);
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+                return null;
+            });
+        }
+
+        public<U> LockedContext<T> scrollAndUpdate(RelationalDao<U> relationalDao, DetachedCriteria criteria, ScrollMode scrollMode, Function<ScrollableResults,List<U>> handler) {
```

Comment:
Rename this to update and change the signature to take a method that updates only one object. ScrollMode is unnecessary as input here. The handler function will be applied to all objects.

Reply:
More details on this on test comment below.